File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        level_queue = deque()
        levels = []
        if root:
            level_queue.append(root)
        while level_queue:
            level = []
            qLen = len(level_queue)
            for i in range(qLen):
                node = level_queue.popleft()                
                if node is not None:
                    level.append(node.val)
                    level_queue.append(node.left)
                    level_queue.append(node.right)
            if level:
                levels.append(level)
        return levels

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        std::queue<TreeNode*> level_queue;
        vector<vector<int>> levels;
        if(root != nullptr) {
            level_queue.push(root);
        }
        while(!level_queue.empty()) {
            int queue_len = level_queue.size();
            vector<int> level;
            while(queue_len) {
                TreeNode* node = level_queue.front();
                level_queue.pop();
                if(node != nullptr) {
                    level_queue.push(node->left);
                    level_queue.push(node->right);
                    level.push_back(node->val);
                }
                queue_len--;
            }
            if(!level.empty()) {
                levels.push_back(level);
            }
        }
        
        return levels;
    }
};